stream/src/public_key/handshake.rs

```diff
@@ -36,6 +38,33 @@ impl<P: PublicKey> Info<P> {
             recipient,
             ephemeral_public_key,
             timestamp,
+            continuation_tag: None,
+        }
+    }
+
+    /// Augment this message by computing a tag over some data.
+    ///
+    /// For example, to allow for the info to also be linked to the hello message sent by the other peer,
+    /// or having this information attest to a MAC using a previous shared session key.
+    pub fn with_tag_over(self, data: &[u8]) -> Self {
+        let mut sha = Sha256::new();
+        sha.update(data);
+        Self {
+            continuation_tag: Some(sha.finalize()),
+            ..self
+        }
+    }
```

Comment:
Are you ever expected to use `with_tag_over` more than once? Should document the behavior here